src/session/queryable.c

```diff
@@ -227,6 +227,14 @@ static z_result_t _z_trigger_queryables_inner(_z_session_rc_t *zsrc, _z_msg_quer
     _z_query_t q =
         _z_query_alias(&msgq->_ext_value, &key, &msgq->_parameters, zsrc, qid, &msgq->_ext_attachment, anyke);
     _z_query_rc_t query = _z_query_rc_new_from_val(&q);
+    if (_Z_RC_IS_NULL(&query)) {
+        return _Z_ERR_SYSTEM_OUT_OF_MEMORY;
+    }
+    // Clear values to avoid double free
+    *q_key = _z_keyexpr_null();
```

Comment:
AFAIS, _z_trigger_queryables_inner called only from _z_trigger_queryables, in this case will better move the code to _z_trigger_queryables and remove cleanup, in this case we don't need this workaround.

Reply:
I'm not sure I understand, the inner thing is akin to a goto to make sure we clean stuff we have ownership of. In the case of trigger_queryable that would be in case of errors or absence of queryables related to the keyexpr.